Source/DafnyCore/Resolver.cs

```diff
@@ -16595,6 +16595,9 @@ public ResolutionContext(ICodeContext codeContext, bool isTwoState)
       MemberDecl member = null;
 
       var name = resolutionContext.InReveal ? "reveal_" + expr.SuffixName : expr.SuffixName;
+      if (!expr.Lhs.WasResolved()) {
+        return null;
+      }
       var lhs = expr.Lhs.Resolved;
       if (lhs != null && lhs.Type is Resolver_IdentifierExpr.ResolverType_Module) {
```

Comment:
I don't understand this part; it looks like we're already checking for `null`; if that's not enough, shouldn't we remove this null check now that you have the early exit?

Reply:
`WasResolved()` on expressions is only equivalent to the type not being `null`.
However, `.Resolved` will still return `null` even if `.WasResolved()` is true, if the underlying expression is e.g. not a `ConcreteSyntaxExpression` (so it has nothing to "resolve"). it's just that `.WasResolved()` is part of the _requires_ of `.Resolved`.

Reply:
Got it, thanks!  And why do we do the check here instead of just after resolving the LHS a few lines above?

Reply:
I think it's better to do the check just before when we need it for sure - e.g. if this code is copy-pasted at some point, this check won't be missed - but I acknowledge it could be subject to debate.